### ISSUE TITLE
Update records by pushing full object into the store

### ISSUE THREAD

**Issue by MichaelVdheeren** [emberjs/data#2053]
Whenever I send data through the push method with partial = true this nevertheless results in the full object being updated IF you push the full data-set. An example:

The current record is similar to:
```
{person: {
  firstName: 'michael'
  lastName: 'vdheeren'
}}
```
the update sent contains:
```
{person: {
  firstName: 'mick'
  lastName: 'vdheeren'
}}
```
As a result the observers on ```lastName``` are also triggered while the value wasn't changed in fact. I suspect that ```record.setupData``` changes all attributes and not only the ones that really changed. Sending partial updates (only the firstName) is not an option as we want to be sure that even if you would miss one of the updates your records are consistent.

I've currently created a suboptimal solution (doesn't work for relationships) to support this, but I don't like the fact that it is suboptimal ;-). Any suggestions? Things that I missed? Can we change the checks to support this by default?

```
update: function(type, data) {
    var id = coerceId(data.id),
        record = this.recordForId(type, id),
        rev = null, value = null;

    // This doesn't support relationship updates yet
    record.beginPropertyChanges();
    record.eachAttribute(function(name) {
      rev   = record._data[name];
      value = data[name];

      if (value && !Em.isEqual(rev,value)) {
        record.set(name, value);
      }
    });
  }
```

**Comment by igorT** [comment]
cc @wecc where are we on this?

**Comment by wecc** [comment]
Fixed by #3216, closing.
